### PR TITLE
LPS-127697 move the logic that allows invalid available locales to DDMFormValidatorImpl

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-127697

### CODE REVIEW

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java [hunk 1]
```diff
@@ -416,13 +416,16 @@ public DataDefinition postSiteDataDefinitionByContentType(
 		DDMForm ddmForm = DataDefinitionDDMFormUtil.toDDMForm(
 			dataDefinition, _ddmFormFieldTypeServicesTracker);
 
+		DataDefinitionContentType dataDefinitionContentType =
+			_dataDefinitionContentTypeTracker.getDataDefinitionContentType(
+				contentType);
+
+		ddmForm.setAllowInvalidAvailableLocalesForProperty(
+			dataDefinitionContentType.allowEmptyDataDefinition());
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java:424]: I think you should call `dataDefinitionContentType.allowInvalidAvailableLocalesForProperty` , rigth?

Comment (marcosapmf) [reply modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java]: Right, @jeyvison. Sorry 😅
I forced a push for this pull with the fix. Thanks!

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:forward

**Pull request closed without merging**